crates/sui/tests/reconfiguration_tests.rs

```diff
@@ -436,90 +439,400 @@ async fn test_reconfig_with_committee_change_basic() {
 
     let mut authorities =
         spawn_test_authorities(genesis_objects.clone().into_iter(), &init_configs).await;
+    tokio::time::sleep(Duration::from_secs(3)).await;
+
+    let _effects = execute_join_committee_txes(
+        &authorities,
+        gas_objects
+            .clone()
+            .into_iter()
+            .map(|obj| obj.compute_object_reference())
+            .collect::<Vec<_>>(),
+        stake.compute_object_reference(),
+        &new_validator,
+        new_node_config,
+    )
+    .await;
+
+    trigger_reconfiguration(&authorities).await;
+    // Check that a new validator has joined the committee.
+    authorities[0].with(|node| {
+        assert_eq!(
+            node.state()
+                .epoch_store_for_testing()
+                .committee()
+                .num_members(),
+            5
+        );
+    });
+
+    let mut new_node_config_clone = new_node_config.clone();
+    // Make sure that the new validator config shares the same genesis as the initial one.
+    new_node_config_clone.genesis = init_configs.validator_configs[0].genesis.clone();
+    let handle = start_node(
+        &new_node_config_clone,
+        RegistryService::new(Registry::new()),
+    )
+    .await;
+    // We have to manually insert the genesis objects since the test utility doesn't.
```

Comment:
i think we can put the objects into genesis though, right?

Reply:
I'll also use the same approach of writing to genesis for the test I introduced once I rebase. This was written before your PR had merged.

Reply:
This is rebased now and we're no longer doing this in my test.